packages/customWidgets/calendar-web/src/components/CalendarContainer.ts

```diff
@@ -197,19 +216,27 @@ export default class CalendarContainer extends Component<Container.CalendarConta
         }
     };
 
+    private async setPeriod(mxObject: MxObject): Promise<void> {
+        const period = await this.getSelectedPeriod(mxObject);
+        this.setState({
+            defaultView: period || this.props.defaultView
```

Comment:
Why call this "defaultView" on the state if it can change?

Reply:
I chose this because it is also used as defaultView and the name in the library is also called defaultView to load a view.